Add KafkaContainer#withListener (#7333)
testcontainers/testcontainers-java: docs/modules/kafka.md

See the [versions interoperability matrix](https://docs.confluent.io/platform/current/installation/versions-interoperability.html) for more details. 

## Register listeners

> There are scenarios where connecting to the broker is needed. The consumer/producer can be another container

Author: I've updated this section

> in the same network or a different process where port to connect to is different from the exposed port `9093`.

Reviewer: ```suggestion
in the same network or a different process where the port to connect differs from the default exposed port `9093`.
```

E.g [Toxiproxy](../../docs/modules/toxiproxy.md).

<!--codeinclude-->

> [Register listener](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:registerListener

Reviewer: ```suggestion
[Register additional listener](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:registerListener
```

<!--/codeinclude-->

Container defined in the same network:

<!--codeinclude-->
[Create kcat container](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:createKCatContainer
<!--/codeinclude-->

> Client using the new listener registered:

Reviewer: ```suggestion
Client using the new registered listener:
```
